### PR TITLE
NCTL :: Join & Leave updates

### PR DESCRIPTION
**Pull request by ghost**
1. Updated join & leave mechanism.

### CODE REVIEW

File: utils/nctl/docs/commands-ctl.md [hunk 1]
```diff
@@ -84,40 +84,40 @@ Attempts to rotate a network's validator set - used to verify joining/leaving sc
 nctl-rotate
 ```
 
-### nctl-start node={X:-all} loglevel={Y:-($RUST_LOG | debug)}
+### nctl-start node={X:-all} hash=Y loglevel={Z:-($RUST_LOG | debug)}
```
Comment (linclelinkpart5) [inline utils/nctl/docs/commands-ctl.md:87]: Neat! That's a nice way of handling the trusted hash juggling

File: utils/nctl/sh/assets/setup.sh [hunk 2]
```diff
@@ -65,30 +66,52 @@ function _set_chainspec()
...
+#######################################
+function _set_chainspec_config()
+{
+    local PATH_TO_NET=${1}
+    local PATH_TO_FILE=${2}
+    local GENESIS_NAME=${3}
+    local GENESIS_TIMESTAMP=${4}
+    local COUNT_MAX_NODES=${5}
+
+    local SCRIPT=(
```
Comment (linclelinkpart5) [inline utils/nctl/sh/assets/setup.sh:99]: Much more robust way of filling in templated values, I like this

**Review by linclelinkpart5 — APPROVED** [review]

File: utils/nctl/docs/commands-ctl.md [hunk 3]
```diff
@@ -84,40 +84,40 @@ Attempts to rotate a network's validator set - used to verify joining/leaving sc
...
 nctl-start
 
 nctl-start node=all loglevel=debug (same as above)
 
-nctl-start node=3 loglevel=info
+nctl-start node=3 hash=<A-TRUSTED_HASH> loglevel=info
 ```
 
-### nctl-start-after-n-blocks node={X:-all} offset={Y:-1} loglevel={Z:-($RUST_LOG | debug)}
+### nctl-start-after-n-blocks node={W:-all} offset={X:-1} hash=Y loglevel={Z:-($RUST_LOG | debug)}
```
Comment (goral09) [inline utils/nctl/docs/commands-ctl.md:99]: I am not sure we will be able to use this command with `hash=…` in practice. It's not possible to know the trusted hash to sync with in advance to starting the network.

File: utils/nctl/docs/commands-ctl.md [hunk 4]
```diff
@@ -84,40 +84,40 @@ Attempts to rotate a network's validator set - used to verify joining/leaving sc
...
-Starts node X after chain has advanced by Y eras.  If X=all then all nodes are started.  
+Starts node W after chain has advanced by X eras.  If W=all then all nodes are started.  If hash is specified then it is injected into node's config prior to spin-up.
 
 ```
 nctl-start-after-n-eras
 
 nctl-start-after-n-eras node=all offset=1  (same as above)
 
-nctl-start-after-n-eras node=8 offset=4
+nctl-start-after-n-eras node=8 offset=4 hash=<A-TRUSTED_HASH>
```
Comment (goral09) [inline utils/nctl/docs/commands-ctl.md:120]: Same as above – will probably be impossible to use in practice.

**Pull request merged**